Marlin/src/module/planner_bezier.cpp

```diff
@@ -190,6 +190,10 @@ void cubic_b_spline(const float position[NUM_AXIS], const float target[NUM_AXIS]
     bez_target[E_AXIS] = interp(position[E_AXIS], target[E_AXIS], t);
     clamp_to_software_endstops(bez_target);
 
+    #if ENABLED(FWRETRACT)
+      planner.apply_retract(bez_target);
+    #endif
```

Comment:
Example: If `apply_retract` were added to `planner.apply_modifiers` then this won't be needed here.

Reply:
>UBL doesn't use apply_leveling or unapply_leveling at this time

This seems to be the one exception to that rule as `apply_leveling` is called a few lines further down if `HAS_UBL_AND_CURVES` is defined.

I agree that combining `apply_leveling` and `apply_retract` is the right thing to do, but I was concerned that it might open a can of worms trying to do so. Perhaps it would be best to keep this PR as is to address the issue at hand (i.e. the print becoming skewed when using firmware retraction) and perform the consolidation in a separate PR.